### PR TITLE
Add wellpicks to known contents

### PR DESCRIPTION
**Pull request by perolavsvendsen**
This PR partly solves #341 and adds the content `wellpicks`.
- Add wellpicks to list of known contents
- Add definitions of known table index columns

### CODE REVIEW

File: schema/definitions/0.8.0/examples/table_wellpicks.yml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  vertical_domain: depth # / time / null
+  depth_reference: msl # / seabed / etc
+  spec: # class/layout dependent, optional? Can spec be expanded to work for all data types?
+    ncol: 281
+    nrow: 441
+    undef: -999.25  # Allow both number and string
+    columns:
+      - BULK_OIL
+      - NET_OIL
+      - PORE_OIL
```
Comment (jcrivenaes) [inline schema/definitions/0.8.0/examples/table_wellpicks.yml:86]: Not columns you would expect for well picks

File: schema/definitions/0.8.0/examples/table_wellpicks.yml [hunk 2]
```diff
@@ -0,0 +1,122 @@
+# Example metadata for an inplace volume table.
+
```
Comment (jcrivenaes) [inline schema/definitions/0.8.0/examples/table_wellpicks.yml:2]: Correct comment

File: schema/definitions/0.8.0/examples/table_wellpicks.yml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  table_index:
+    - WELL
+    - HORIZON
+  is_prediction: true # A mechanism for separating pure QC output from actual predictions
+  is_observation: false
+  description:
+    - Inplace volumes calculated on MyGrid
+
+display:
+  name: Inplace, MyGrid
```
Comment (jcrivenaes) [inline schema/definitions/0.8.0/examples/table_wellpicks.yml:96]: Same issue here

File: src/fmu/dataio/_definitions.py [hunk 4]
```diff
@@ -59,12 +59,14 @@ def __post_init__(self):
...
     "khproduct": None,
     "timeseries": None,
+    "wellpicks": None,
 }
 
 STANDARD_TABLE_INDEX_COLUMNS = {
     "inplace_volumes": ["ZONE", "REGION", "FACIES", "LICENCE"],
     "timeseries": ["DATE"],  # summary
     "rft": ["measured_depth", "well", "time"],
+    "wellpicks": ["WELL", "HORIZON"],
```
Comment (jcrivenaes) [inline src/fmu/dataio/_definitions.py:69]: I would think that MD is (more or less) required? (I have seen well picks with X Y Z in rare cases, but then the software need to compute MD)

**Review by jcrivenaes — CHANGES_REQUESTED** [review]
See comments

File: schema/definitions/0.8.0/examples/table_wellpicks.yml (hunk 1 above)
Comment (perolavsvendsen) [reply schema/definitions/0.8.0/examples/table_wellpicks.yml]: Yes, good catch, thanks. Ctrl+c/Ctrl+v error... Sorted out now 👍

File: schema/definitions/0.8.0/examples/table_wellpicks.yml (hunk 2 above)
Comment (perolavsvendsen) [reply schema/definitions/0.8.0/examples/table_wellpicks.yml:2]: Fixed

File: schema/definitions/0.8.0/examples/table_wellpicks.yml (hunk 3 above)
Comment (perolavsvendsen) [reply schema/definitions/0.8.0/examples/table_wellpicks.yml]: Fixed

File: src/fmu/dataio/_definitions.py (hunk 4 above)
Comment (perolavsvendsen) [reply src/fmu/dataio/_definitions.py:69]: Yes, correct. Added the `MD` column to the test data. However, the `STANDARD_TABLE_INDEX_COLUMNS` is referring to (categorical) index columns, which would still be just "WELL" and "HORIZON" in this case. So line `69` should not change (?)

**Pull request merged**